src/d3a/models/area/__init__.py

```diff
@@ -190,23 +191,29 @@ def _cycle_markets(self, _trigger_event=True, _market_cycle=False, deactivate=Fa
                 and _trigger_event and ConstSettings.BalancingSettings.ENABLE_BALANCING_MARKET:
             self.dispatcher.broadcast_balancing_market_cycle()
 
-    def tick(self, is_root_area=False):
+    def tick(self):
         if ConstSettings.IAASettings.MARKET_TYPE == 2 or \
                 ConstSettings.IAASettings.MARKET_TYPE == 3:
             if ConstSettings.GeneralSettings.EVENT_DISPATCHING_VIA_REDIS:
                 self.dispatcher.publish_market_clearing()
             else:
                 for market in self.all_markets:
                     market.match_offers_bids()
+
         self.events.update_events(self.now)
-        if self.current_tick % self.config.ticks_per_slot == 0 and is_root_area:
-            self._cycle_markets()
-        self.dispatcher.broadcast_tick()
         self.current_tick += 1
         if self._markets:
             for market in self._markets.markets.values():
                 market.update_clock(self.current_tick)
 
+    def tick_and_dispatch(self):
```

Comment:
This looks much better now (in comparison to calling the same code in the tick method), thanks for changing that.